src/Controller/GoogleAuthController.php

```diff
@@ -74,45 +101,101 @@ public static function create(ContainerInterface $container) {
+
+    $this->dataHandler->set('oauth2state', $state);
 
-    return new RedirectResponse($client->createAuthUrl());
+    return new TrustedRedirectResponse($google_login_url);
   }
 
   /**
-   * Callback function to login user.
+   * Response for path 'user/login/google/callback'.
+   *
+   * Google returns the user here after user has authenticated in Google.
    */
   public function callback() {
-    /* @var \Google_Client $client */
-    $client = $this->networkManager->createInstance('social_auth_google')->getSdk();
+    // Checks if user cancel login via Google.
+    $error = $this->request->getCurrentRequest()->get('error');
+    if ($error == 'access_denied') {
+      drupal_set_message($this->t('You could not be authenticated.'), 'error');
+      return $this->redirect('user.login');
+    }
 
-    $this->googleManager->setClient($client)->authenticate();
+    /* @var \League\OAuth2\Client\Provider\Google false $google */
+    $google = $this->networkManager->createInstance('social_auth_google')->getSdk();
 
-    // Saves access token so that event subscribers can call Google API.
-    $this->session->set('social_auth_google_access_token', $this->googleManager->getAccessToken());
+    // If Google client could not be obtained.
+    if (!$google) {
+      drupal_set_message($this->t('Social Auth Google not configured properly. Contact site administrator.'), 'error');
+      return $this->redirect('user.login');
+    }
 
-    // Gets user information.
-    $user = $this->googleManager->getUserInfo();
-    // If user information could be retrieved.
-    if ($user) {
-      return $this->userManager->authenticateUser($user->getEmail(), $user->getName(), $user->getId(), $user->getPicture());
+    $state = $this->dataHandler->get('oauth2state');
+
+    // Retrieves $_GET['state'].
+    $retrievedState = $this->request->getCurrentRequest()->query->get('state');
+    if (empty($retrievedState) || ($retrievedState !== $state)) {
+      $this->userManager->nullifySessionKeys();
+      drupal_set_message($this->t('Google login failed. Unvalid oAuth2 State.'), 'error');
+      return $this->redirect('user.login');
     }
 
-    drupal_set_message($this->t('You could not be authenticated, please contact the administrator'), 'error');
-    return $this->redirect('user.login');
+    // Saves access token to session.
+    $this->dataHandler->set('access_token', $this->googleManager->getAccessToken());
+
+    $this->googleManager->setClient($google)->authenticate();
+
+    // Gets user's info from Google API.
+    if (!$google_profile = $this->googleManager->getUserInfo()) {
+      drupal_set_message($this->t('Google login failed, could not load Google profile. Contact site administrator.'), 'error');
+      return $this->redirect('user.login');
+    }
+
+    // Store the data mapped with data points define is
+    // social_auth_google settings.
+    $data = [];
+
+    if (!$this->userManager->checkIfUserExists()) {
```

Comment:
You're not passing any of the required arguments to the method. Are you testing the code?